packages/core/src/shared/clients/codecatalystClient.ts

```diff
@@ -175,7 +178,16 @@ export async function createClient(
 ): Promise<CodeCatalystClient> {
     const sdkClient = await createCodeCatalystClient(connection, regionCode, endpoint, retryOptions)
     const c = new CodeCatalystClientInternal(connection, sdkClient)
-    await c.verifySession()
+    try {
+        await c.verifySession()
+    } catch (e) {
+        if (!(e instanceof ToolkitError) || e.code !== 'AccessDeniedException') throw e
```

Comment:
should we also check `isAwsError`? https://github.com/aws/aws-toolkit-vscode/blob/e5fdf4109207f801a9b91a4d162ef1fb8d1bffd1/packages/core/src/codecatalyst/auth.ts#L314

Reply:
isAwsError ensures that the error contains a `time` attribute. `ToolKitError` doesn't contain a time. Now that you mention it, I don't think the `isOnboardingError` should even work at all. I'll test and confirm the behavior and update `isOnboardingError` if that is the case.